### PR TITLE
Back-end work for user profile.

### PR DESCRIPTION
**Pull request by akatsoulas**
@johngian r?

### CODE REVIEW

File: feedthefox/base/templates/jinja2/profile.html [hunk 1]
```diff
@@ -9,20 +9,30 @@
...
-        <p>
-          <span class="glyphicon glyphicon-envelope" aria-hidden="true"></span>
-
-        </p>
+        {% if user.photo %}
+          <img src="{{ user.photo }}"
+               alt="{{ user.get_full_name() }}" class="img-responsive img-circle">
+        {% else %}
+          <img src="{{ static('img/avatar.jpg') }}"
+               alt="{{ user.get_full_name() }}" class="img-responsive img-circle">
```
Comment (comzeradd) [inline feedthefox/base/templates/jinja2/profile.html:17]: Shouldn't we use `settings.DEFAULT_AVATAR` here?

File: feedthefox/base/templates/jinja2/profile.html [hunk 2]
```diff
@@ -61,12 +71,16 @@
...
-        <form>
+        <form id="newsletter-form" method="post">
+          {% csrf_token %}
           <div class="form-group">
-            <input type="checkbox" name="privacy" value="1" checked>
-            I agree to receive regular emails (at least once a week) from
-            Mozilla’s Foxfooding Community Manager to provide updates on the
-            program and communicate new tasks.
+            <p>
+              {{ newsletter_form.receive_newsletter }}
```
Comment (comzeradd) [inline feedthefox/base/templates/jinja2/profile.html:78]: Just a minor note, that we need this to be checked by default.

Comment (akatsoulas) [inline feedthefox/base/templates/jinja2/profile.html:78]: The default value in the model it's `True`

**Comment by akatsoulas** [comment]
@comzeradd pr updated

File: feedthefox/settings.py [hunk 3]
```diff
@@ -9,6 +9,11 @@
 """
 
 import os
+try:
+    from urllib.parse import urljoin
```
Comment (johngian) [inline feedthefox/settings.py]: Do we need this? There is no reference to `urljoin` in `settings.py`.

File: feedthefox/users/views.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+def view_profile(request):
+    user = request.user
+    newsletter_form = forms.ReceiveNewsLetterForm(request.POST or None,
+                                                  instance=user)
+
+    if newsletter_form.is_valid():
+        newsletter_form.save()
+        return redirect('view_profile')
+
+    data = {'user': user,
```
Comment (johngian) [inline feedthefox/users/views.py]: We already have `django.contrib.auth.context_processors.auth` in `context_processors` so `request.user` is already in the context.

**Comment by johngian** [comment]
~~`Github` provider is missing from installed apps.~~

File: feedthefox/settings.py (hunk 3 above)
Comment (akatsoulas) [inline feedthefox/settings.py]: Indeed, I was using at first but when I removed it I forgot this reference. Nice catch

**Comment by akatsoulas** [comment]
@comzeradd @johngian pr updated

**Comment by comzeradd** [comment]
Looks good. r+ :coffee:

**Pull request merged**